### ISSUE TITLE
False positive for Lint/DuplicateMethods in anonymous classes

### ISSUE THREAD

**Issue by jcoyne** [rubocop/rubocop#15058]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/DuplicateMethods

### Steps to reproduce the problem

1. create `example.rb` with the following:
```ruby
# frozen_string_literal: true

# Example of violation
class Example
  [
    Class.new do
      def initialize
        @value = nil
      end
    end,

    Class.new do
      def initialize
        @value = nil
      end
    end
  ]
end
```

2. run rubocop:

```
example.rb:13:7: W: Lint/DuplicateMethods: Method Example::Object#initialize is defined at both example.rb:7 and example.rb:13.
      def initialize
      ^^^^^^^^^^^^^^
```

### Expected behavior

Anonymous classes should not be counted by Lint/DuplicateMethods. They are separate methods.

### Actual behavior

It is finding an issue.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
rubocop -V
1.86.0 (using Parser 3.3.10.2, rubocop-ast 1.49.1, analyzing as Ruby 3.1, running on ruby 4.0.1) [arm64-darwin25]
  - rubocop-capybara 2.22.1
  - rubocop-rspec 3.9.0
  - rubocop-rails 2.34.3
```

**Comment by koic** [comment]
This is a duplicate issue with #15056.
